src/cryptenroll/cryptenroll-tpm2.c

```diff
@@ -257,8 +257,19 @@ int enroll_tpm2(struct crypt_device *cd,
                         return log_error_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE), "TPM2 seal/unseal verification failed.");
         }
 
+        if (use_pin) {
+                /* To prevent using a secret from a compromised TPM without the pin, append salted pin.
+                 * See following attack on AMD's fTPMs: https://arxiv.org/abs/2304.14717 */
```

Comment:
I might be missing something (I know very little about TPM), but why append the salted pin? The salted PIN is passed to the TPM (hopefully not stored there). Why not append the non-salted pin? This way the encryption key is built with data that was not passed to the TPM, thus making TPM+Passphrase to be at least secure as the Passphrase.